### ISSUE TITLE
T-SQL: Do not indent after OUTER APPLY statements

### ISSUE THREAD

**Issue by borchero** [sqlfluff/sqlfluff#3685]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

Similarly to #3672, there is an issue with SQLFluff treating anything after an `OUTER APPLY` clause as nested join (which should not happen).

### Expected Behaviour

```sql
SELECT table1.*
FROM table1
OUTER APPLY table2
-- This should not be indented
INNER JOIN table3
    ON table1.col = table3.col
```

### Observed Behaviour

```sql
SELECT table1.*
FROM table1
OUTER APPLY table2
    -- This part is indented
    INNER JOIN table3
        ON table1.col = table3.col
```

### How to reproduce

See SQL statement above.

### Dialect

T-SQL

### Version

`1.2.1`

### Configuration

```
[sqlfluff]
dialect = tsql
exclude_rules = L044

[sqlfluff:rules:L010]
capitalisation_policy = upper

[sqlfluff:rules:L030]
extended_capitalisation_policy = upper
```

### Are you willing to work on and submit a PR to address the issue?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by borchero** [comment]
**help wanted**
